modules/ROOT/pages/tuning-backpressure-maxconcurrency.adoc

```diff
@@ -0,0 +1,29 @@
+= Back-Pressure and Max Concurrency
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Under heavy load, Mule runtime engine (Mule) might not have resources available to handle a specific event. This issue might occur because all threads are busy and the current flow’s concurrency has been exceeded already. +
+In case Mule cannot handle an event, Mule logs a message about the condition: +
+`Flow 'flowName' is unable to accept new events at this time` +
+Also known as back-pressure. Mule back-pressure actions are specific to each connector’s source.
+
+When there is a producer component that outputs information at a higher rate than a consumer component can handle, you can use the flow’s `maxConcurrency` parameter to set the maximum number of concurrent messages that a flow can process at any given time. +
+This scenario can manifest in various forms, as those components could be internal/external, flows/sources, etc.
+
+Although Mule applies back-pressure on its own, the `maxConcurrency` parameter helps to keep under control the performance of your flows, especially when a backend server may affect when having high concurrency. See how to set the parameter in the following example:
```

Comment:
See previous suggestions.

Do we need to mention and, if so, explain how Mule applies solutions to back pressure (applies SOLUTIONS, right, not applies the back pressure itself?)

```suggestion
Although Mule automatically applies internal methods to control back pressure, you can customize the `maxConcurrency` parameter to help control flow performance of your flows:
```
Again, I think that this is wrong: Although Mule **applies back-pressure** on its own,
and this does not make sense: "especially when a backend server may affect when having high concurrency"
Does this mean that a backend server can cause high concurrency or increase existing high concurrency?